src/fairseq2/nn/utils/gradient.py

```diff
@@ -75,7 +75,7 @@ def forward(ctx: Any, x: Tensor, scale: float) -> Tensor:  # type: ignore[overri
 
         ctx.scale = scale
 
-        return x.detach().clone().requires_grad_(True)
```

Comment:
Why do we detach and clone here? Why not just return x? This removes x from the autograd graph right, why is it necessary?